File: src/frontend/src/types/schedule.ts
export enum ScheduleType {
  RuleExecution = 0,
  DataProfiling = 1,
}

export enum ScheduleExecutionStatus {
  Running = 0,
  Completed = 1,
  Failed = 2,
}

export interface Schedule {
  id: string;
  name: string;
  type: ScheduleType;
  ruleId?: string;
  workspaceId?: string;
  datasetName?: string;
  tableName?: string;
  cronExpression: string;
  isEnabled: boolean;
  lastRunAt?: string;
  nextRunAt?: string;
  createdAt: string;
  createdBy: string;
}

export interface ScheduleExecution {
  id: string;
  scheduleId: string;
  startedAt: string;
  completedAt?: string;
  status: ScheduleExecutionStatus;
  resultSummary?: string;
  errorMessage?: string;
}

export interface CreateScheduleRequest {
  name: string;
  type: ScheduleType;
  ruleId?: string;
  workspaceId?: string;
  datasetName?: string;
  tableName?: string;
  cronExpression: string;
  isEnabled: boolean;
}

File: src/backend/ClarityDQ.Core/Entities/Schedule.cs
namespace ClarityDQ.Core.Entities;

public class Schedule
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public ScheduleType Type { get; set; }
    public Guid? RuleId { get; set; }
    public Rule? Rule { get; set; }
    public string? WorkspaceId { get; set; }
    public string? DatasetName { get; set; }
    public string? TableName { get; set; }
    public string CronExpression { get; set; } = string.Empty;
    public bool IsEnabled { get; set; }
    public DateTime? LastRunAt { get; set; }
    public DateTime? NextRunAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public string CreatedBy { get; set; } = string.Empty;
}

public enum ScheduleType
{
    RuleExecution,
    DataProfiling
}

public class ScheduleExecution
{
    public Guid Id { get; set; }
    public Guid ScheduleId { get; set; }
    public Schedule? Schedule { get; set; }
    public DateTime StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public ScheduleExecutionStatus Status { get; set; }
    public string? ResultSummary { get; set; }
    public string? ErrorMessage { get; set; }
}

public enum ScheduleExecutionStatus
{
    Running,
    Completed,
    Failed
}
